src/abstractdataframe/iteration.jl

```diff
@@ -459,12 +459,12 @@ julia> mapcols(x -> x.^2, df, cols=r"y")
 """
 function mapcols(f::Union{Function, Type}, df::AbstractDataFrame; cols=All())
     if cols === All() || cols === Colon()
-        apply = fill(true, ncol(df))
+        apply = Iterators.repeated(true)
     else
         picked = Set(names(df, cols))
         apply = Bool[name in picked for name in names(df)]
+        @assert length(apply) == ncol(df)
```

Comment:
Any reason why this could be false?

Reply:
No - it is just that later we use `zip` which does not check this invariant.

If you prefer I can remove it.

Reply:
Well it seems to me that the comprehension that creates `apply` already guarantees that invariant (unless `names` is broken but then we have more serious problems...).

Reply:
OK - I removed the `@assert`